lib/image_optim/handler.rb

```diff
@@ -38,6 +39,9 @@ def process
       @src ||= @original
       @dst ||= @original.temp_path
 
+      FileUtils.chmod(@cache_dir_mode, @src) unless @cache_dir_mode.nil?
```

Comment:
also had to set the permissions here, i think because temp_path was giving mode `600` files